iiif_prep/iiif_prep/archivesspace.py

```diff
@@ -0,0 +1,40 @@
+from asnake.aspace import ASpace
+
+
+class NoResultsError(Exception):
+    pass
+
+
+class MultipleResultsError(Exception):
+    pass
+
+
+class ArchivesSpaceClient:
+    def __init__(self, baseurl, username, password, repository):
+        self.client = ASpace(
+            baseurl=baseurl,
+            username=username,
+            password=password,
+            repository=repository).client
+        self.repository = repository
+
+    def get_diary_refid(self, diary):
+        refid = None
+        search_url = "repositories/{}/search?page=1&type[]=archival_object&q=digital_object:+{}.pdf".format(
+            self.repository, diary)
+        results_page = self.client.get(search_url).json()
+        if len(results_page.get("results")) == 1:
+            refid = results_page.get("results")[0].get("ref_id")
+        elif len(results_page.get("results")) == 0:
+            raise NoResultsError("0 results found for {}".format(diary))
+        elif len(results_page.get("results")) == 2:
+            if results_page.get("results")[0].get(
+                    "ref_id") == results_page.get("results")[0].get("ref_id"):
+                refid = results_page.get("results")[0].get("ref_id")
```

Comment:
Not sure I understand what's going on here...

Reply:
I'll add a comment, but this is a way to deal with duplicate DAOs. (I know in the case of officers' diaries, if there are multiple DAOs attached to a refid, they were inadvertently ingested multiple times. There should be some clean up, but that shouldn't block the IIIF pipeline.)

Reply:
OK, but:
1) your equality statement will always evaluate to True because you're checking if the first value is equal to itself.
2) should probably log something here if cleanup is needed.